tests/system/test_system_async.py

```diff
@@ -0,0 +1,1000 @@
+# Copyright 2017 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import asyncio
+import datetime
+import math
+import pytest
+import operator
+import os
+import re
+
+from google.oauth2 import service_account
+
+from google.api_core.exceptions import AlreadyExists
+from google.api_core.exceptions import FailedPrecondition
+from google.api_core.exceptions import InvalidArgument
+from google.api_core.exceptions import NotFound
+from google.cloud._helpers import _datetime_to_pb_timestamp
+from google.cloud._helpers import UTC
+from google.cloud import firestore_v1 as firestore
+from test_utils.system import unique_resource_id
+
+FIRESTORE_CREDS = os.environ.get("FIRESTORE_APPLICATION_CREDENTIALS")
+FIRESTORE_PROJECT = os.environ.get("GCLOUD_PROJECT")
+RANDOM_ID_REGEX = re.compile("^[a-zA-Z0-9]{20}$")
+MISSING_DOCUMENT = "No document to update: "
+DOCUMENT_EXISTS = "Document already exists: "
+UNIQUE_RESOURCE_ID = unique_resource_id("-")
+
+_test_event_loop = asyncio.new_event_loop()
+pytestmark = pytest.mark.asyncio
+
+
+@pytest.fixture(scope=u"module")
+def client():
+    credentials = service_account.Credentials.from_service_account_file(FIRESTORE_CREDS)
+    project = FIRESTORE_PROJECT or credentials.project_id
+    yield firestore.AsyncClient(project=project, credentials=credentials)
+
+
+@pytest.fixture
+async def cleanup():
+    operations = []
+    yield operations.append
+
+    for operation in operations:
+        await operation()
```

Comment:
do each of these `await`s need to be guarded for possible exceptions? We probably don't want to let one `operation` failing preventing the rest from being attempted.

Reply:
This fixture is used to delete documents and collections after the test is run, so I believe failing here is acceptable. The awaits only happen after the test is run, and failure should not directly impact other tests.

Reply:
sounds good